app/src/main/java/cm/aptoide/pt/account/view/store/ManageStoreFragment.java

```diff
@@ -305,147 +503,195 @@ public void bindViews(View view) {
+    twitterEndRowIcon = (ImageView) view.findViewById(R.id.edit_store_twitter_plus);
+    youtubeRow = view.findViewById(R.id.edit_store_youtube);
+    youtubeTextAndPlus =
+        (RelativeLayout) view.findViewById(R.id.edit_store_youtube_text_plus_wrapper);
+    youtubeUsernameWrapper =
+        (CustomTextInputLayout) view.findViewById(R.id.edit_store_youtube_username_wrapper);
+    youtubeUser = (EditText) view.findViewById(R.id.edit_store_youtube_username);
+    youtubeText = (TextView) view.findViewById(R.id.edit_store_youtube_title);
+    youtubeEndRowIcon = (ImageView) view.findViewById(R.id.edit_store_youtube_plus);
 
     waitDialog = GenericDialogs.createGenericPleaseWaitDialog(getActivity(),
         getApplicationContext().getString(R.string.please_wait_upload));
     toolbar = (Toolbar) view.findViewById(R.id.toolbar);
   }
 
-  private ViewModel updateAndGetStoreModel() {
-    currentModel = ViewModel.update(currentModel, storeName.getText()
+  private ManageStoreViewModel updateAndGetStoreModel() {
+    currentModel = ManageStoreViewModel.update(currentModel, storeName.getText()
         .toString(), storeDescription.getText()
         .toString());
     currentModel.setStoreTheme(themeSelectorAdapter.getSelectedTheme());
+    currentModel.setStoreLinks(getStoreLinks());
     return currentModel;
   }
 
-  private void setupViewsDefaultDataUsingCurrentModel() {
+  /**
+   * This method will add a object to the returned list for each social channel already defined by
+   * the user.
+   *
+   * @return
+   */
+  private List<SimpleSetStoreRequest.StoreLinks> getStoreLinks() {
+    List<SimpleSetStoreRequest.StoreLinks> storeLinksList = new ArrayList<>();
+    if (!TextUtils.isEmpty(facebookUser.getText()
+        .toString())) {
+      storeLinksList.add(
+          new SimpleSetStoreRequest.StoreLinks(FACEBOOK, setFacebookUrl(facebookUser.getText()
+              .toString())));
+    }
+    if (!TextUtils.isEmpty(twitchUser.getText()
+        .toString())) {
+      storeLinksList.add(new SimpleSetStoreRequest.StoreLinks(Store.SocialChannelType.TWITCH,
+          setTwitchUrl(twitchUser.getText()
+              .toString())));
+    }
+    if (!TextUtils.isEmpty(twitterUser.getText()
+        .toString())) {
+      storeLinksList.add(new SimpleSetStoreRequest.StoreLinks(Store.SocialChannelType.TWITTER,
+          setTwitterUrl(twitterUser.getText()
+              .toString())));
+    }
+    if (!TextUtils.isEmpty(youtubeUser.getText()
+        .toString())) {
+      storeLinksList.add(new SimpleSetStoreRequest.StoreLinks(Store.SocialChannelType.YOUTUBE,
+          setYoutubeUrl(youtubeUser.getText()
+              .toString())));
+    }
+    if (storeLinksList.isEmpty()) {
+      return Collections.emptyList();
+    }
+    return storeLinksList;
+  }
+
+  private String setYoutubeUrl(String youtubeUsername) {
+    if (youtubeUsername.contains("http")) {
```

Comment:
avoid this! use the URL [Pattern matcher from Android](https://developer.android.com/reference/android/util/Patterns.html#WEB_URL)

Reply:
again, multiple methods can be avoided by using the enum SocialChannelType